MapboxCoreNavigation/PortableRouteController.swift

```diff
@@ -67,6 +67,24 @@ open class PortableRouteController: RouteController {
         return !offRoute
     }
     
+    /**
+     Advances the leg index. This override also advances the leg index of the native navigator.
+     
+     This is a convienence method provided to advance the leg index of any given router without having to worry about the internal data structure of the router.
+     */
+    override public func advanceLegIndex(location: CLLocation) {
+        precondition(!routeProgress.isFinalLeg, "Can not increment leg index beyond final leg.")
+        
+        super.advanceLegIndex(location: location)
+        
+        let status = navigator.getStatusForTimestamp(location.timestamp)
+        let legIndex = status.legIndex
+        let routeIndex = status.routeIndex
+
+        //The first route is the active one in the navigator.
+        navigator.changeRouteLeg(forRoute: routeIndex, leg: legIndex + 1)
```

Comment:
Can we do this to keep the models verifiably synchronized? Or is there a possibility that the navigator’s leg index is non-zero-based?

```suggestion
        navigator.changeRouteLeg(forRoute: routeIndex, leg: routeProgress.legIndex)
```